crates/filament/src/ir_passes/lower/build_ctx.rs

```diff
@@ -237,7 +234,17 @@ impl<'a> BuildCtx<'a> {
                 );
             };
             let delay = delay.concrete(self.comp);
-            let typ = FsmType::new(states, delay, self.disable_slow_fsms);
+            let typ = match self.comp.attrs.get_numeric(ast::NumAttr::SlowFSM) {
+                None => FsmType::Simple(states),
+                Some(0) => FsmType::Simple(states), // 0 is a special case for simple FSMs
+                Some(n) => {
+                    assert!(
```

Comment:
Can we move this into the well-formedness check earlier in the compiler pipeline?

Reply:
Ok, will do but I will also keep this assert just in case things go wrong somewhere in the compiler.

Reply:
Ah yes, this is the correct choice!

Reply:
Sure, that also seems reasonable!